subiquity/server/controllers/filesystem.py

```diff
@@ -377,11 +380,59 @@ async def guided_GET(self, wait: bool = False) -> GuidedStorageResponse:
             disks=[labels.for_client(d, min_size=min_size) for d in disks],
             storage_encryption=se)
 
+    def _add_structure(self, *, disk: Disk, next_offset: int, is_last: bool,
+                       structure: snapdapi.VolumeStructure):
+        ptype = structure.type.split(',', 1)[1].upper()
+        if structure.offset is not None:
+            offset = structure.offset
+        else:
+            offset = next_offset
+        if structure.role == snapdapi.Role.SYSTEM_DATA and is_last:
+            size = gaps.largest_gap(disk).size
+        else:
+            size = structure.size
```

Comment:
Do we have checks at some point that there is enough room?

Reply:
no. i guess we should filter the disks we present to only show ones that the gadget will git on.

Reply:
Yes, that would go a long way.  For now we can assume reformatting the entire disk and not do side-by-side, correct?

Reply:
now done

Reply:
> Yes, that would go a long way. For now we can assume reformatting the entire disk and not do side-by-side, correct?

weellll there is an incoming wibble about recovery partitions but for now lets say yes

Reply:
That concerns me less, that's a predictable size section as opposed to coping with arbitrary user disk layout.